### PR TITLE
Find source for some special components

### PR DESCRIPTION
**Pull request by hanchuntao**
[OLCS-292](https://issues.redhat.com/browse/OLCS-292)

Fixes OLCS-292

### CODE REVIEW

**Comment by hanchuntao** [comment]
@rhyw @qduanmu @huiwangredhat PTAL

**Comment by huiwangredhat** [comment]
LGTM, at least it's ok for the found special cases.

File: openlcs/libs/sc_handler.py [hunk 1]
```diff
@@ -333,6 +371,12 @@ def get_remote_source_path(self, component, extra_src_dir):
                     else:
                         source_path = path
                         break
+        else:
+            # Add patterns for some special components that name, version
+            # mismatch with source path in source container. More detail,
+            # check PELC-4511, CLOUDBLD-3809, and OLCS-292.
```
[the remark this answers is not in the document]
Comment (hanchuntao) [reply openlcs/libs/sc_handler.py:377]: OK, I will add more detail here. I added before, but there are a lot of data here about components, so I added in the [OLCS-292](https://issues.redhat.com/browse/OLCS-292).

[the remark this answers is not in the document]
Comment (hanchuntao) [reply openlcs/libs/sc_handler.py:402]: updated

**Pull request merged**